src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritConnection.java

```diff
@@ -94,6 +94,7 @@
     private AuthenticationUpdater authenticationUpdater = null;
     private final Set<ConnectionListener> listeners = new CopyOnWriteArraySet<ConnectionListener>();
     private int sshRxBufferSize = SSH_RX_BUFFER_SIZE;
+    private StringBuilder eventBuffer = null;
```

Comment:
What about all the other instance variables. Are they all threadsafe? The instances of StringBuilder will never be accessed by more than 1 thread, the one that owns the instance. Threadsafety means that an object behaves well when 2 threads access the same object. But how could this happen here? When we have multiple instances of GerritConnection that run in parallel each holds its own instance of a StringBuilder.